File: solutions/2574. Left and Right Sum Differences/2574.cpp
// Time complexity: O(n)
// Space complexity: O(n)
class Solution {
 public:
  vector<int> leftRightDifference(const vector<int>& nums) {
    vector<int> ans;
    int leftSum = 0;
    int rightSum = accumulate(nums.begin(), nums.end(), 0);

    for (const int num : nums) {
      rightSum -= num;
      ans.push_back(abs(leftSum - rightSum));
      leftSum += num;
    }

    return ans;
  }
};

File: solutions/2574. Left and Right Sum Differences/2574.py
# Time complexity: O(n)
# Space complexity: O(n)
class Solution:
  def leftRightDifference(self, nums: List[int]) -> List[int]:
    ans = []
    leftSum = 0
    rightSum = sum(nums)

    for num in nums:
      rightSum -= num
      ans.append(abs(leftSum - rightSum))
      leftSum += num

    return ans  